File: aoj/volume100/ocaml/A10025.ml
let pi = 4.0 *. atan 1.0

let solve a b deg =
    let rad = deg *. pi /. 180.0 in
    let c   = sqrt (a**2.0 +. b**2.0 -. 2.0*.a*.b*.(cos rad)) in

    let area      = a*.b*.(sin rad) /. 2.0 in
    let perimeter = a +. b +. c in
    let h         = 2.0*.area /. a in

    Printf.printf "%.6f\n%.6f\n%.6f\n" area perimeter h

let () =
    Scanf.scanf "%f %f %f" solve

File: aoj/volume100/haskell/A10025.hs
import Control.Applicative
import Text.Printf

main = do
  [a, b, deg] <- fmap (read :: String -> Double) . words <$> getLine
  (\(area, perimeter, h) -> printf "%.6f %.6f %.6f" area perimeter h :: IO ())
    $ solve a b deg

solve a b deg =
  (area, perimeter, h)
  where
    rad       = deg * pi / 180
    c         = sqrt (a**2.0 + b**2.0 - 2.0*a*b*(cos rad))
    area      = a*b*(sin rad) / 2
    perimeter = a + b + c
    h         = 2*area / a
